activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -23,6 +23,20 @@ def table_alias_for(table_name)
         table_name[0...table_alias_length].tr('.', '_')
       end
 
+      # Returns the relations names usuable to back Active Record models.
+      # Defaults to all #tables and #views.
```

Comment:
@yui-knk it's not so much about wether the user can overwrite these methods but the adapters can. We can't say for sure that every adapter will have working views and tables as data source.